File: exercises/scala/001_request-response-pattern/src/main/scala/dev/fResult/akka/coffeeMachine/CoffeeMachineCommand.scala
package dev.fResult.akka.coffeeMachine

import akka.actor.typed.ActorRef
import dev.fResult.akka.Coffee
import dev.fResult.akka.barista.BaristaCommand

enum CoffeeMachineCommand {
  case BrewCoffee(coffee: Coffee, replyTo: ActorRef[BaristaCommand])
  case PickupCoffee(coffee: Coffee)
}

File: exercises/java/003_timers/src/main/java/com/akka/training/coffeeMachine/CoffeeMachineCommand.java
package com.akka.training.coffeeMachine;

import akka.actor.typed.ActorRef;
import com.akka.training.Coffee;

public sealed interface CoffeeMachineCommand
    permits CoffeeMachineCommand.BrewCoffee,
        CoffeeMachineCommand.CoffeeReady,
        CoffeeMachineCommand.PickupCoffee {

  record BrewCoffee(Coffee coffee, ActorRef<CoffeeReady> replyTo) implements CoffeeMachineCommand {}

  record CoffeeReady(Coffee coffee) implements CoffeeMachineCommand {}

  /*
   * Represents the Barista picking up the coffee and resetting the coffee machine, so that it's ready for the next coffee
   */
  static final class PickupCoffee implements CoffeeMachineCommand {}
}
